ntfysh_client/NotificationDialog.cs

```diff
@@ -0,0 +1,263 @@
+
+                    this.progressBar1.Visible = true;
+                    this.lbTimeout.Visible = true;
+                    this._timeout = timeout_ms;
+                } else
+                {
+                    this.progressBar1.Visible = false;
+                    this.lbTimeout.Visible = false;
+                }
+            }
+            else
+            {
+                this.progressBar1.Visible = false;
+                this.lbTimeout.Visible = false;
+            }
+
+            // ok, show the window
+            this.Show();
+            this.SetWindowPosition();
+        }
+
+        private void UpdateProgress(object? sender, EventArgs e)
+        {
+            if (this.shownStopwatch == null)
+            {
+                return;
+            }
+            this.progress = (int)((this._timeout - this.shownStopwatch.ElapsedMilliseconds) * 100 / this._timeout);
+            this.lbTimeout.Text = $"{(int)(this._timeout - this.shownStopwatch.ElapsedMilliseconds) / 1000}";
+        }
+
+        protected override void SetVisibleCore(bool value)
+        {
+            this.SetWindowPosition();
+            if (value)
+            {
+                this.BringToFront();
+                AnimateWindow(
+                    this.Handle,
+                    time: 250,
+                    flags: NFWinUserAnimateWindowConstnats.AW_SLIDE | NFWinUserAnimateWindowConstnats.AW_VER_NEGATIVE
+                );
+            }
+            base.SetVisibleCore(value);
+        }
+
+        protected override void OnShown(EventArgs e)
+        {
+            base.OnShown(e);
+        }
+
+        private void SetWindowPosition()
+        {
+            int workingtop = Screen.PrimaryScreen.WorkingArea.Height - this.Height;
+            this.Top = workingtop - NotificationDialog.ScreenMargin;
+
+            int workingleft = Screen.PrimaryScreen.WorkingArea.Width - this.Width;
+            this.Left = workingleft - NotificationDialog.ScreenMargin;
+        }
+
+        private void ui_hide_window(object? sender, EventArgs? e)
+        {
+            AnimateWindow(
+                this.Handle,
+                time: 250,
+                flags: NFWinUserAnimateWindowConstnats.AW_SLIDE | NFWinUserAnimateWindowConstnats.AW_VER_POSITIVE | NFWinUserAnimateWindowConstnats.AW_HIDE
+            );
+            this.IsVisible = false;
+        }
+
+        private void handleTimeout(object? sender, EventArgs? e)
+        {
+            this.cancelTimer();
+            if (this.InvokeRequired)
+            {
+                // on a background thread, so invoke on the UI thread
+                this.Invoke(new Action(() => this.ui_hide_window(sender, e)));
+            }
+            else
+            {
+                // in the UI thread, invoke directly
+                this.ui_hide_window(sender, e);
+            }
+        }
+
+        private Image? ConvertToolTipIconToImage(ToolTipIcon icon)
+        {
+            switch (icon)
```

Comment:
This is a great candidate for the new `switch expression` syntax, and then it can be turned into a lambda

```csharp
private Image? ConvertToolTipIconToImage(ToolTipIcon icon) => icon switch
{
    ToolTipIcon.Info => SystemIcons.Information.ToBitmap(),
    ToolTipIcon.Warning => SystemIcons.Warning.ToBitmap(),
    ToolTipIcon.Error => SystemIcons.Error.ToBitmap(),
    _ => null
}
```

Reply:
updated